cpp/oneapi/dal/detail/error_messages.cpp

```diff
@@ -0,0 +1,156 @@
+MSG(neighbor_count_lt_one, "Neighbor count lower than one")
+
+/* Jaccard */
+MSG(column_begin_gt_column_end, "Column begin is greater than column end")
+MSG(empty_edge_list, "Empty edge list")
+MSG(negative_interval, "Negative interval")
+MSG(interval_gt_vertex_count, "Interval is greater than vertex count")
+MSG(row_begin_gt_row_end, "Row begin is greater than row end")
+
+/* PCA */
+MSG(component_count_lt_zero, "Component count is lower than zero")
+MSG(input_data_cc_lt_desc_component_count,
+    "Input data column count is lower than component count provided in descriptor")
+MSG(input_model_eigenvectors_cc_neq_input_data_cc,
+    "Input model eigenvectors column count is not equal to input data column count")
+MSG(input_model_eigenvectors_rc_neq_desc_component_count,
+    "Eigenvectors' row count in input model is not equal to component count provided in descriptor")
+MSG(pca_svd_based_method_is_not_implemented_for_gpu,
+    "PCA SVD-based method is not implemented for GPU")
+
+/* SVM */
+MSG(c_leq_zero, "C is lower than or equal to zero")
+MSG(cache_size_leq_zero, "Cache size lower that or equal to zero")
+MSG(input_kernel_is_empty, "Input kernel function is empty")
+MSG(input_model_coeffs_are_empty, "Input model coeffs are empty")
+MSG(input_model_coeffs_rc_neq_input_model_support_vector_count,
+    "Input model coeffs row count is not equal to support vector count provided in input model")
+MSG(input_model_does_not_match_kernel_function, "Input model does not match kernel function type")
+MSG(input_model_support_vectors_are_empty, "Input model support vectors are empty")
+MSG(input_model_support_vectors_cc_neq_input_data_cc,
+    "Input model support vectors column count is not equal to input data column count")
+MSG(input_model_support_vectors_rc_neq_input_model_support_vector_count,
+    "Support vectors row count is not equal to support vector count in input model")
+MSG(sigma_leq_zero, "Sigma lower than or equal to zero")
+MSG(svm_smo_method_is_not_implemented_for_gpu, "SVM SMO method is not implemented for GPU")
+MSG(tau_leq_zero, "Tau is lower than or equal to zero")
+
+/* Kernel Functions */
+MSG(input_y_is_empty, "Input y is empty")
+MSG(input_x_is_empty, "Input x is empty")
+MSG(input_x_cc_neq_y_cc, "Input x column count is not qual to y column count")
+
+/* Decision Forest */
+MSG(bootstrap_is_incompatible_with_error_metric,
+    "Values of bootstrap and error metric parameters provided "
+    "in descriptor are incompatible to each other")
+MSG(bootstrap_is_incompatible_with_variable_importance_mode,
+    "Values of bootstrap and variable importance mode parameters provided "
+    "in descriptor are incompatible to each other")
+MSG(decision_forest_train_dense_method_is_not_implemented_for_gpu,
+    "Decision forest train dense method is not implemented for GPU")
+MSG(decision_forest_train_hist_method_is_not_implemented_for_cpu,
+    "Decision forest train hist method is not implemented for CPU")
+
```

Comment:
`#undef MSG` doesn't matter cause it's cpp file?

Reply:
yes